client/pages/MyProfile.jsx

```diff
@@ -22,39 +23,42 @@ export default class MyProfile extends React.Component {
         const { error } = result;
         if (!error) {
           this.setState({ post: result });
+          fetch('/api/username', {
+            headers: {
+              'Content-Type': 'application/json',
+              'x-access-token': token
+            }
+          })
+            .then(res => res.json())
+            .then(result => {
+              const [data] = result;
+              const { username } = data;
+              this.setState({ username });
+            });
```

Comment:
I don't think this fetch is required. You can decode the token from localStorage. If you update the server, you can put the username in the token payload.